### PR TITLE
Fix build problems with a fresh forked k-9

### PR DESCRIPTION
**Pull request by pueffl**
At the moment, when you checkout K-9 from scratch and install the latest
Android Studio, there are severeral problems with `gradle build`:

- removed uses-sdk from AndroidManifest.xml
- updated buildTools to 28.0.3
- updated kotlin to 1.2.71
- set Gradle jvmargs to prevent from OutOfMemory
- set targetSdk to 26 as AS sees all <26 as an Error because of
  Google Play.
- upgrade to latest Gradle 4.x version

I also excluded 2 Files from packaging as they are created more than
twice. Don't know why this happens and it should be fixed sometime but
as the files are empty I hope this is ok.

Please ensure that your pull request meets the following requirements - thanks !

* Follows our existing [codestyle](https://github.com/k9mail/k-9/wiki/CodeStyle).
* Does not break any unit tests.
* Contains a reference to the issue that it fixes.
* For cosmetic changes add one or multiple images, if possible.

### CODE REVIEW

**Comment by philipwhiuk** [comment]
The license fails because the travis.yml file needs updating at the same time as the build.gradle.

We can't blindly update the target SDK however - it will break the app as @wiktor-k mentions.

**Comment by cketti** [comment]
While we don't want to change the targetSdkVersion just yet, this PR contains other changes we do want.

File: app/k9mail/build.gradle [hunk 1]
```diff
@@ -36,7 +36,7 @@ dependencies {
 
 android {
     compileSdkVersion buildConfig.compileSdk
-    buildToolsVersion buildConfig.buildTools
```
Comment (cketti) [inline app/k9mail/build.gradle]: Please don't change this line. Changing the value of the variable is sufficient.

File: app/k9mail/build.gradle [hunk 2]
```diff
@@ -46,7 +46,7 @@ android {
         versionName '5.700-SNAPSHOT'
 
         minSdkVersion buildConfig.minSdk
-        targetSdkVersion 23
+        targetSdkVersion 26
```
Comment (cketti) [inline app/k9mail/build.gradle]: As mentioned before, please revert this change.

File: app/k9mail/build.gradle [hunk 3]
```diff
@@ -94,6 +94,8 @@ android {
         exclude 'META-INF/NOTICE.txt'
         exclude 'META-INF/README'
         exclude 'LICENSE.txt'
+        exclude 'META-INF/imap_debug.kotlin_module'
```
Comment (cketti) [inline app/k9mail/build.gradle]: Let's go a step further and add this:
```
        exclude "META-INF/proguard/**"
        exclude 'META-INF/*.version'
        exclude 'META-INF/*.kotlin_module'
        exclude "kotlin/**"
```

File: gradle.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+org.gradle.jvmargs=-Xmx2g -XX:MaxPermSize=2048m -XX:+HeapDumpOnOutOfMemoryError
```
Comment (cketti) [inline gradle.properties]: This doesn't seem to be necessary to run Gradle from the command line. I'd rather not include it because this will probably override my Android Studio config. 
If building vanilla K-9 Mail with a freshly installed Android Studio is really a problem I'd rather add a note to the README that tells developers to adjust their Android Studio's config. It also might not be an issue anymore with the recently released Android Studio 3.3.

File: build.gradle [hunk 5]
```diff
@@ -3,11 +3,11 @@ buildscript {
         buildConfig = [
                 'compileSdk': 27,
                 'minSdk': 19,
-                'buildTools': '27.0.3'
+                'buildTools': '28.0.3'
```
Comment (cketti) [inline build.gradle]: Please also change `.travis.yml` to install the new build tools version. That should fix the CI build in this pull request.
